### DISCUSSION TITLE
how to deploy cobbler with  a docker container ?

### DISCUSSION THREAD

**Discussion by wangxu186189** [cobbler/cobbler#3545] [Q&A] [answered]
how to deploy cobbler with  a docker container ?

**Comment by SchoolGuy** [comment, marked as answer]
So far nothing from our org is published. There is an issue for this already open https://github.com/cobbler/cobbler/issues/2453. Still, since Cobbler manages the services through systems and similar manners, this is sadly not as easy as "building an image" since we need to prepare the application to restart services that are unavailable via a CLI command.

However, there is https://github.com/PrincetonUniversity/pni-cobbler which was reported working a while ago. I haven't tried it personally.

I will not try or accept a solution based on a "fat" container upstream since this is simply just a VM in the end. We use this for years already in unit and integration testing and it is not suited for production use.

**Reply by wangxu186189** [reply]
ok, many thanks.
